### PR TITLE
3.0: Adding buildspec to build docker in CodeBuild and other minor enhancements

### PR DESCRIPTION
**Pull request by demartinofra**
* The buildspec file can be used to create a CodeBuild projects that builds the ParallelCluster docker image and uploads it to ECR
* Temporarily adding AdministratorAccess to the API infrastructure sample in order to enable testing
* Restructure lambda entrypoint for better XRay tracing
* Add filename, line number and function name to log records

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: api/docker/awslambda/buildspec.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+version: 0.2
+
+# Variables to set
```
Comment (lukeseawalker) [inline api/docker/awslambda/buildspec.yml:3]: minor - would be nice to add checks for required parameters

File: api/docker/awslambda/buildspec.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# GIT_REF: branch name or commit id to build
+# IMAGE_REPO_NAME: name of the ECR repo where the image is pushed
+# IMAGE_TAG: tag for the Docker images
+phases:
+  pre_build:
+    commands:
+      - echo Logging in to Amazon ECR...
+      - aws ecr get-login-password --region $AWS_DEFAULT_REGION | docker login --username AWS --password-stdin $ECR_ENDPOINT
+      - echo Cloning ParallelCluster GitHub repository...
+      - git clone https://github.com/aws/aws-parallelcluster aws-parallelcluster && cd aws-parallelcluster && git checkout $GIT_REF
```
Comment (lukeseawalker) [inline api/docker/awslambda/buildspec.yml:14]: is `GIT_REF` always required?

File: api/docker/awslambda/buildspec.yml (hunk 1 above)
Comment (demartinofra) [reply api/docker/awslambda/buildspec.yml]: done

File: api/docker/awslambda/buildspec.yml (hunk 2 above)
Comment (demartinofra) [reply api/docker/awslambda/buildspec.yml:18]: added a default

**Review by lukeseawalker — APPROVED** [review]

**Pull request merged**